src/App.jsx

```diff
@@ -14,33 +15,105 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sort, setSort] = useState([...goodsFromServer]);
+  const [reverse, setReverse] = useState(false);
+  const [active, setActive] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSort = good => {
+    let newSort;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (good) {
+      case 'abc':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.localeCompare(a));
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.localeCompare(b));
+          setActive(good);
+        }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        break;
+      case 'length':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.length - a.length);
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.length - b.length);
+          setActive(good);
+        }
+
+        break;
+      case 'reverse':
+        if (active === 'abc') {
+          setReverse(!reverse);
+        } else if (active === 'reverse') {
+          setReverse(!reverse);
+          setActive('');
+        } else if (active === 'length') {
+          setReverse(!reverse);
+        } else {
+          setReverse(!reverse);
+          setActive('reverse');
```

Comment:
The logic for toggling the reverse state could be simplified. Currently, it checks the `active` state multiple times to determine whether to toggle `reverse` and set `active`. Consider refactoring this logic to make it more concise and easier to understand.